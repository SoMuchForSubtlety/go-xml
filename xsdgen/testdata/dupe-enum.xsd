<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
  xmlns="http://example.org/" targetNamespace="http://example.org/">

  <xs:element name="TypeA">
    <xs:complexType>
      <xs:sequence>
        <xs:element name="dType">
          <xs:simpleType>
            <xs:annotation>
              <xs:documentation>type for weekday</xs:documentation>
            </xs:annotation>
            <xs:restriction base="xs:string">
              <xs:enumeration value="MON" />
              <xs:enumeration value="TUE" />
              <xs:enumeration value="WED" />
              <xs:enumeration value="THU" />
              <xs:enumeration value="FRI" />
              <xs:enumeration value="SAT" />
              <xs:enumeration value="SUN" />
            </xs:restriction>
          </xs:simpleType>
        </xs:element>
      </xs:sequence>
    </xs:complexType>
  </xs:element>

  <xs:element name="TypeB">
    <xs:complexType>
      <xs:sequence>
        <xs:element name="DType">
          <xs:simpleType>
            <xs:annotation>
              <xs:documentation>type for weekday</xs:documentation>
            </xs:annotation>
            <xs:restriction base="xs:string">
              <xs:enumeration value="MON" />
              <xs:enumeration value="TUE" />
              <xs:enumeration value="WED" />
              <xs:enumeration value="THU" />
              <xs:enumeration value="FRI" />
              <xs:enumeration value="SAT" />
              <xs:enumeration value="SUN" />
            </xs:restriction>
          </xs:simpleType>
        </xs:element>
      </xs:sequence>
    </xs:complexType>
  </xs:element>

</xs:schema>
