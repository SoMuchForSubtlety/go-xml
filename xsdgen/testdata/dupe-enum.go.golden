// Code generated by xsdgen.test. DO NOT EDIT.

package ws

// type for weekday
// May be one of MON, TUE, WED, THU, FRI, SAT, SUN
type DType string

const (
	DType_MON DType = "MON"
	DType_TUE DType = "TUE"
	DType_WED DType = "WED"
	DType_THU DType = "THU"
	DType_FRI DType = "FRI"
	DType_SAT DType = "SAT"
	DType_SUN DType = "SUN"
)

type TypeA struct {
	DType DType `xml:"http://example.org/ dType"`
}

type TypeB struct {
	DType DType `xml:"http://example.org/ DType"`
}
